src/did/src/send_raw_transaction.rs

```diff
@@ -0,0 +1,486 @@
+};
+use crate::transaction::AccessList;
+use crate::{Bytes, Transaction, H160, U256};
+
+/// `send_raw_transaction` request payload
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize)]
+pub struct SendRawTransactionRequest {
+    /// The signature of the transaction
+    pub signature: Signature,
+    /// The transaction data
+    pub transaction: TransactionData,
+}
+
+impl TryFrom<Transaction> for SendRawTransactionRequest {
+    type Error = &'static str;
+
+    fn try_from(value: Transaction) -> Result<Self, Self::Error> {
+        let signature = Signature {
+            r: value.r,
+            s: value.s,
+            v: value.v,
+        };
+
+        let data = match value
+            .transaction_type
+            .map(|x| x.as_u64())
+            .unwrap_or(TRANSACTION_TYPE_LEGACY)
+        {
+            TRANSACTION_TYPE_LEGACY => TransactionData::Legacy(Legacy {
+                chain_id: value.chain_id,
+                nonce: value.nonce,
+                gas_price: value.gas_price.ok_or("Gas price is missing")?,
+                gas_limit: value.gas,
+                to: value.to,
+                value: value.value,
+                input: value.input.into(),
+            }),
+            TRANSACTION_TYPE_EIP1559 => TransactionData::Eip1559(Eip1559 {
+                chain_id: value.chain_id,
+                nonce: value.nonce,
+                gas_limit: value.gas,
+                max_fee_per_gas: value.max_fee_per_gas.ok_or("Max fee per gas is missing")?,
+                max_priority_fee_per_gas: value
+                    .max_priority_fee_per_gas
+                    .ok_or("Max priority fee per gas is missing")?,
+                to: value.to,
+                value: value.value,
+                access_list: value.access_list.ok_or("Access list is missing")?,
+                input: value.input.into(),
+            }),
+            TRANSACTION_TYPE_EIP2930 => TransactionData::Eip2930(Eip2930 {
+                chain_id: value.chain_id.ok_or("Chain id is missing")?,
+                nonce: value.nonce,
+                gas_price: value.gas_price.ok_or("Gas price is missing")?,
+                gas_limit: value.gas,
+                to: value.to,
+                value: value.value,
+                access_list: value.access_list.ok_or("Access list is missing")?,
+                input: value.input.into(),
+            }),
+            _ => return Err("Unknown transaction type"),
+        };
+
+        Ok(SendRawTransactionRequest {
+            signature,
+            transaction: data,
+        })
+    }
+}
+
+/// Transaction type and data
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize)]
+pub enum TransactionData {
+    Legacy(Legacy),
+    Eip2930(Eip2930),
+    Eip1559(Eip1559),
+}
+
+/// Transaction input
+#[derive(Debug, Clone, PartialEq, Eq, CandidType, Serialize, Deserialize, Default)]
+pub enum TransactionInput {
```

Comment:
This issue is my fault since I provided a wrong example in our last chat, sorry for that. Anyway, the correct enum should be like:
```
pub enum TxKind {
    /// A transaction that creates a contract.
    Create,
    /// A transaction that calls a contract or transfer.
    Call(H160), <-- this contains an address
}
```
then you should use it as the `to` field of a Transaction